pkg/controller/backingstore/backingstore_controller.go

```diff
@@ -66,3 +68,19 @@ func Add(mgr manager.Manager) error {
 
 	return nil
 }
+
+// ModeChangedPredicate will only allow events that changed Status.Mode.ModeCode.
+// This predicate should be used only for BackingsStore objects!
+type ModeChangedPredicate struct {
+	predicate.Funcs
+}
+
+// Update implements the update event trap for LabelsChangedPredicate
+func (p ModeChangedPredicate) Update(e event.UpdateEvent) bool {
+	if e.ObjectOld == nil || e.ObjectNew == nil {
+		return false
```

Comment:
What does false mean? I didn't read the entire context but wanted to make sure this was intentional - after returning false here are we still queueing this backing store to reconcile?

Reply:
false means we drop the event - not queueing. 

For Update events, both new and old objects must be non-nil. (and also new and old meta)

Please note that the first time a resource is introduced into the client cache a create event (not update) will be raised, even if the resource has been created a long time ago. This means that whenever the controller is initialized (when the operator is initializing) all watched resources will get a create even dispatched for them and because we do not filter any create events, in this case, a reconcile will be queued.

The checks here are a safeguard that Jacly has copied from the predicates that I wrote and I, in turn, copied from the built-in predicates.

Reply:
false mean don't process:
https://godoc.org/sigs.k8s.io/controller-runtime/pkg/predicate#Predicate